lib/WebSocket.js

```diff
@@ -848,7 +848,7 @@ function establishConnection(ReceiverClass, SenderClass, socket, upgradeHead) {
   self._receiver.onerror = function onerror(reason, errorCode) {
     // close the connection when the receiver reports a HyBi error code
     self.close(typeof errorCode !== 'undefined' ? errorCode : 1002, '');
-    self.emit('error', reason, errorCode);
+    self.emit('error', new Error(reason));
```

Comment:
to be safe i would add a type check here for `reason`. it might be already an instance of error. only do `new Error()` when it's a string